lom/src/plugins/plugins_common/plugins_helper.go

```diff
@@ -464,3 +467,366 @@ func GetResponse(request *lomipc.ActionRequestData, anomalyKey string, response
+        return
+    }
+
+    /* Publish heartbeat only after above validations pass.*/
+    pluginHeartBeat := PluginHeartBeat{PluginName: subscriptionBasedPluginUtil.PluginName, EpochTime: time.Now().Unix()}
+    hbchan <- pluginHeartBeat
+}
+
+/*
+ * handleRequest handles a subscription request.
+ *
+ * Parameters:
+ * - request: A pointer to an ActionRequestData struct. This is the request to handle.
+ *
+ * The function initializes a backoffTimeSecs timer and a flag to indicate if the connection to the gnmi server needs to be restarted.
+ *
+ * This function does not return a value.
+ */
+func (subscriptionBasedPluginUtil *SubscriptionBasedPluginUtil) handleRequest(request *lomipc.ActionRequestData) {
+    subscriptionBasedPluginUtil.pluginLogger.LogInfo("Subscription handler initialized for plugin (%s)", subscriptionBasedPluginUtil.PluginName)
+    // Initialize the backoffTimeSecs timer
+    // The backoffTimeSecs timer is used to reconnect to the gnmi server in case of connection errors
+    retryCount := 0
+    maxDelay := float64(subscriptionBasedPluginUtil.backoffTimeSecs)
+    var backoffTimer *time.Timer
+
+    // Flag to indicate if the connection to gnmi server needs to be restarted.
+    // This flag is set to true when there is connection error
+    restartConnection := false
+
+loop:
+    for {
+        var errCh chan error
+        var goroutineResponseCh chan *lomipc.ActionResponseData
+
+        // Call the plugin request function in a separate goroutine
+        if backoffTimer == nil {
+            errCh = make(chan error, 1)
+            goroutineResponseCh = make(chan *lomipc.ActionResponseData, 1)
+
+            lomcommon.GetGoroutineTracker().Start(plugin_prefix+subscriptionBasedPluginUtil.PluginName+"_plugin_request_"+GetUniqueID(),
+                func() {
+                    response, err := subscriptionBasedPluginUtil.requestFunc(request, subscriptionBasedPluginUtil.ctx, restartConnection)
+                    if err != nil {
+                        errCh <- err
+                    } else {
+                        // successful detection.
+                        goroutineResponseCh <- response
+                    }
+                })
+        }
+
+        select {
+        case data := <-goroutineResponseCh:
+            // If a shutdown has been initiated, stop processing updates
+            if subscriptionBasedPluginUtil.shutDownInitiated.Load() {
+                break loop
+            }
+
+            // successful detection. If a response is received, send the response to the response channel
+            subscriptionBasedPluginUtil.responseChannel <- data
+            subscriptionBasedPluginUtil.numOfConsecutiveErrors.Store(0)
+            restartConnection = false
+            retryCount = 0
+            backoffTimer = nil
+            return
+
+        case err := <-errCh:
+            //unhealthy execution
+            subscriptionBasedPluginUtil.numOfConsecutiveErrors.Add(1)
```

Comment:
Add as todo comment ; when there is max retries, need a way to stop and return error. 
Ensure engine does not call again after request method fails.

Reply:
added